lab_3/main.py

```diff
@@ -17,38 +17,87 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+    checks = [
+        isinstance(text, str),
+        text != "",
+    ]
+
+    if not all(checks) or not any(letter.isalpha() for letter in text):
```

Comment:
`letter.isalpha() for letter in text` is both memory and time consuming. Try to find something else.

Reply:
in process

Reply:
`any(letter.isalpha() for letter in text)` is almost the same - it will check every single letter.